morpheus/_lib/src/io/loaders/rest.cpp

```diff
@@ -17,21 +17,295 @@
+                            const std::string& content_type,
+                            const std::string& body,
+                            const std::unordered_map<std::string, std::string>& x_headers,
+                            int max_retry)
+{
+    std::string ep(endpoint);
+    std::string query;
+    std::string host;
+    std::string target;
+    {
+        py::gil_scoped_acquire gil;
+
+        py::module_ urllib  = py::module::import("urllib.parse");
+        py::object ep_pystr = py::str(ep);
+
+        // Following the syntax specifications in RFC 1808, urlparse recognizes a netloc only if it is properly
+        // introduced by ‘//’. Otherwise the input is presumed to be a relative URL and thus to start with a path
+        // component. ref: https://docs.python.org/3/library/urllib.parse.html Workaround here is to prepend "//" if
+        // endpoint does not already include one
+        if (ep_pystr.attr("find")("//").cast<int>() == -1)
+        {
+            ep = "//" + ep;
+        }
+
+        py::object result = urllib.attr("urlparse")(ep);
+        host              = result.attr("hostname").is_none() ? "" : result.attr("hostname").cast<std::string>();
+        target            = result.attr("path").is_none() ? "" : result.attr("path").cast<std::string>();
+        if (target.empty())
+        {
+            target = "/";
+        }
+        query = result.attr("query").is_none() ? "" : result.attr("query").cast<std::string>();
+
+        // if params exists, overrides query included in endpoint
+        if (!params.empty())
+        {
+            query = params;
+        }
+    }
+
+    net::io_context ioc;
+    tcp::resolver resolver(ioc);
+    beast::tcp_stream stream(ioc);
+    try
+    {
+        auto const endpoint_results = resolver.resolve(host, PORT);
+        auto verb                   = get_http_verb(method);
+        auto request = construct_request(verb, target, query, host, method, content_type, body, x_headers);
+
+        try_get_data(stream, endpoint_results, request, response, max_retry, 1000);
+
+    } catch (std::exception const& e)
+    {
+        VLOG(30) << "Error when calling RESTDataLoader::load(): " << e.what();
+    }
+}
+
+void create_dataframe_from_http_response(http::response<http::dynamic_body>& response,
+                                         py::object& dataframe,
+                                         py::module_& mod_cudf,
+                                         const std::string& strategy)
+{
+    std::string df_json_str = beast::buffers_to_string(response.body().data());
+    boost::algorithm::trim(df_json_str);
+
+    // When calling cudf.read_json() with engine='cudf', it expects an array object as input.
+    // The workaround here is to add square brackets if the original data is not represented as an array.
+    // Submitted an issue to cudf: https://github.com/rapidsai/cudf/issues/13527.
+    if (!(df_json_str.front() == '[' && df_json_str.back() == ']'))
```

Comment:
Does this work for patterns like this?: `\s*[.*]\s*`

We likely want to strip any leading whitespace, carriage returns, etc..